src/io/file_descriptor.windows.cr

```diff
@@ -194,90 +144,44 @@ class IO::FileDescriptor
   end
 
   private def unbuffered_read(slice : Bytes)
-    count = slice.size
-    loop do
-      bytes_read = LibC.read(@fd, slice.pointer(count).as(Void*), count)
-      if bytes_read != -1
-        return bytes_read
-      end
-
-      if Errno.value == Errno::EAGAIN
-        wait_readable
-      else
-        raise Errno.new "Error reading file"
-      end
-    end
-  ensure
-    if (readers = @readers) && !readers.empty?
-      add_read_event
+    overlapped = GC.malloc(sizeof(LibWindows::Overlapped)).as(LibWindows::Overlapped*)
```

Comment:
It was dynamically allocated because it will need to outlive the read_file/write_file call. (The call will return and the actual operation will be completed later.) But I just realized that the current running fiber will be paused until it completes, so allocating on the stack is safe here. I'll change that, thanks for pointing out.